packages/trader/src/Stores/Modules/Trading/Helpers/contract-type.ts

```diff
@@ -512,15 +607,17 @@ export const ContractType = (() => {
         };
     };
 
-    const getCancellation = (contract_type, cancellation_duration, symbol) => {
-        const arr_cancellation_range =
+    const getCancellation = (contract_type: string, cancellation_duration: string, symbol: string) => {
+        const arr_cancellation_range: string[] =
             getPropertyValue(available_contract_types, [contract_type, 'config', 'cancellation_range']) || [];
 
-        const regex = new RegExp('^([0-9]+)|([a-zA-Z]+)$', 'g');
-        const getText = str => {
-            const [duration, unit] = str.match(regex);
+        const regex = /(^\d{1,})|([a-zA-Z]{1,}$)/g;
```

Comment:
due to sonarcloud warning, the following changes to this regex have been made:

1. used a regular expression literal instead of the 'RegExp' constructor.
2. replaced + with {1,} equivalent in order to prevent the regex from being vulnerable to denial of service (DoS) attacks due to backtracking.
3. used concise character class syntax '\d' instead of '[0-9]'.
4. grouped parts of the regex together to make the intended operator precedence explicit